lib/clearbit/enrichment/company.rb

```diff
@@ -0,0 +1,40 @@
+module Clearbit
+  module Enrichment
+    class Company < Base
+      endpoint 'https://company.clearbit.com'
+      path '/v1/companies'
+
+      def self.find(values)
+        unless values.is_a?(Hash)
+          values = {:id => values}
+        end
+
+        if domain = values.delete(:domain)
+          response = get(uri(:domain, domain), values)
+
+        elsif id = values.delete(:id)
+          response = get(id, values)
+
+        else
+          raise ArgumentError, 'Invalid values'
+        end
+
+        if response.status == 202
+          self.new(pending: true)
```

Comment:
My assumption here is that `Base` extends `Mash`? How do you feel about adding an new class to the mix: `Result` which can extend Mash?

```ruby
Result.new(pending: true)
```